### PR TITLE
Automatically cancel previous CI builds

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds a new GitHub actions workflow for canceling previous CI builds when rapidly pushing commits. We recently adding something similar over in `dask/dask` which has helped reduce our CI queue wait time (xref https://github.com/dask/dask/pull/7348)

cc @jakirkham

### CODE REVIEW

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Looking for automerge, but am not seeing it. Do I need to be doing something more?

Screen Shot 2021-03-11 at 10 59 11 AM

**Comment by jrbourbeau** [comment]
Hrm I don't see it either. I suspect this is because we don't have any required branch protection rules...

Let me try adding a rule that a review is required and see if the auto-merge button shows up

**Comment by jakirkham** [comment]
Hmm...still seeing the same thing. Tried refreshing as well

**Comment by jrbourbeau** [comment]
Alright I added a branch protection rule to `main` where at least 1 review approval is required. Unfortunately I don't see the auto-merge button here, though I do see it over in https://github.com/dask/distributed/pull/4579. What do you see here and over in that other PR?

**Comment by jakirkham** [comment]
Ah you are right! Me too 😄 🎉

Maybe there is a security measure to disallow automerge on PRs changing GH Actions? 😉

**Comment by jrbourbeau** [comment]
I'm wondering if it has to do with all the branch protection rules (in this case a single review approval) being met. Would you mind approving https://github.com/dask/distributed/pull/4582 to test that hypothesis out?

**Comment by jakirkham** [comment]
Appears whether one has approved the PR has an effect. Going to try getting rid of my approval here to see what happens

**Comment by jakirkham** [comment]
Nope seems auto-merge as an option is gone

Screen Shot 2021-03-11 at 11 24 24 AM

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks James! 😄 Went ahead and merged the old fashioned way.

There was one failure, but that seems unrelated to these changes. Raised as issue ( https://github.com/dask/distributed/issues/4583 )

**Pull request merged**
